impl/src/metadata.rs

```diff
@@ -272,6 +273,9 @@ impl<'config> MetadataFetcher for CargoInternalsMetadataFetcher<'config> {
         features.insert(feature.clone(), features_or_dependencies.clone());
       }
 
+      // Cargo use SHA256 for checksum so we can use them directly
+      let sha256 = package.manifest().summary().checksum().and_then(|x| Some(x.to_string()));
```

Comment:
Two Nits:

s/`and_then(|x| Some(x.to_string()))`/`map(ToString::to_string)`

The `and_then` is only necessary when the mapping function yields an option. `map` is similar but without the option. `ToString::to_string` is the direct reference to the method (though `|x| x.to_string()` is totally fine as well).

Reply:
Done.